src/main/scala/org/ergoplatform/ErgoBox.scala

```diff
@@ -44,26 +47,24 @@ import scala.runtime.ScalaRunTime
   *                            This height is declared by user and should not exceed height of the block,
   *                            containing the transaction with this box.
   */
-class ErgoBox private(
-                       override val value: Long,
-                       override val ergoTree: ErgoTree,
-                       override val additionalTokens: Seq[(TokenId, Long)] = Seq(),
-                       override val additionalRegisters: Map[NonMandatoryRegisterId, _ <: EvaluatedValue[_ <: SType]] = Map(),
-                       val transactionId: ModifierId,
-                       val index: Short,
-                       override val creationHeight: Int
-                     ) extends ErgoBoxCandidate(value, ergoTree, creationHeight, additionalTokens, additionalRegisters) {
+class ErgoBox(
+         override val value: Long,
+         override val ergoTree: ErgoTree,
+         override val additionalTokens: Coll[(TokenId, Long)] = Colls.emptyColl[(Coll[Byte], Long)],
+         override val additionalRegisters: Map[NonMandatoryRegisterId, _ <: EvaluatedValue[_ <: SType]] = Map(),
+         val transactionId: Coll[Byte],
+         val index: Short,
+         override val creationHeight: Int
+       ) extends ErgoBoxCandidate(value, ergoTree, creationHeight, additionalTokens, additionalRegisters) {
 
   import ErgoBox._
 
-  lazy val id: BoxId = ADKey @@ Blake2b256.hash(bytes)
-
-  override def dataSize: Long = bytes.length
+  lazy val idBytes: BoxId = ADKey @@ Blake2b256.hash(bytes)
```

Comment:
I think type `BoxId` should also be renamed (or just deleted) in such a case.
Also I propose to name this fields as `adKey`, at least it would be consistent with the type